miso-web/src/main/webapp/scripts/hot_dilution.js

```diff
@@ -199,6 +199,56 @@ HotTarget.dilution = {
 
     HotUtils.makeParents('librarydilution', HotUtils.relationCategoriesForDetailed().concat([HotUtils.relations.library()])), 
     HotUtils.makeChildren('librarydilution',[HotUtils.relations.pool()])];
+  },
+
+  confirmSave: function(flatObjects, isCreate, config, table) {
+    var deferred = jQuery.Deferred();
+    
+    dilutions = table.getDtoData();
+    
+    var seen = {};
+    libraries = dilutions.filter(function(dilution){
+      return !(Utils.validation.isEmpty(dilution.volumeUsed) || dilution.volumeUsed <= 0);
+    }).map(function(dilution){
+      return dilution.library;
+    }).filter(function(library){
+      return library != null;
+    }).filter(function(library){
+      return seen.hasOwnProperty(library.id) ? false : (seen[library.id] = true);
+    }).map(function(library){
+      return jQuery.extend(true, {}, library);
+    });
+    
+    if(libraries.length == 0){
+      deferred.resolve();
+      return deferred.promise();
+    }
+            
+    dilutions.filter(function(dilution){
+      return dilution.library != null && dilution.library.volume != null && dilution.volumeUsed != null;
+    }).forEach(function(dilution){
+      libraries.find(function(library){
+        return library.id == dilution.library.id;
+      }).volume -= dilution.volumeUsed;
+    });
+    
+    overUsedCount = libraries.filter(function(library){
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Add the "let", "const" or "var" keyword to this declaration of "overUsedCount" to make it explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)